### PR TITLE
Potential Code-Refactor and Bug Fixes

### PR DESCRIPTION
**Pull request by HarshCasper**
## Description

Summary of fixes:

- Fix check for empty string
- Combine multiple `append`'s into a single call
- Added .deepsource.toml

## Importance

I ran a DeepSource Analysis on my fork of this repository. You can see all the issues raised by DeepSource [here](https://deepsource.io/gh/HarshCasper/fyne/).

DeepSource helps you to automatically find and fix issues in your code during code reviews. This tool looks for anti-patterns, bug risks, performance problems, and raises issues. There are plenty of other issues in relation to Bug Discovery and Anti-Patterns which you would be interested to take a look at.

If you do not want to use DeepSource to continuously analyze this repo, I'll remove the `.deepsource.toml` from this PR and you can merge the rest of the fixes. If you want to setup DeepSource for Continuous Analysis, I can help you set that up.

### CODE REVIEW

File: .deepsource.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version = 1
+
+test_patterns = ["test/**"]
+
+[[analyzers]]
+name = "go"
+enabled = true
+
+  [analyzers.meta]
+  import_paths = ["github.com/HarshCasper/fyne"]
```
Comment (andydotxyz) [inline .deepsource.toml:10]: I don't know if we want to use deepsource on every commit - but if we did this line would be wrong.

**Review by andydotxyz — CHANGES_REQUESTED** [review]
These are helpful changes thanks (one issue noted).
I don't know much about deepsource, but am not sure I see enough in this change log to have another service running against our code for each commit.

Happy to hear other @fyne-io/fyne team thoughts

**Review by Jacalz — CHANGES_REQUESTED** [review]
Thanks for the fixes. The code changes are really nice. As for the deepsource stuff, I am not so sure that we want it included in the repository. I think it would be better to add the few rules that are fixed here (multiple appends after each other and length of string checks) to be part of [staticcheck](https://github.com/dominikh/go-tools) instead, which we use in our CI tests for commits and pull requests.

**Comment by Jacalz** [comment]
I have now opened feature requests for staticcheck to catch the mentioned checks.

**Review by Jacalz — CHANGES_REQUESTED** [review]
@HarshCasper Again, thanks for the PR. It is much appreciated. Do you think that you could remove the `.deepsource.toml`? After that, I think we should be able to get this landed :)

File: .deepsource.toml (hunk 1 above)
Comment (HarshCasper) [reply .deepsource.toml:10]: I will update this up if there are plans to integrate DeepSource.

**Comment by HarshCasper** [comment]
> Thanks for the fixes. The code changes are really nice. As for the deepsource stuff, I am not so sure that we want it included in the repository. I think it would be better to add the few rules that are fixed here (multiple appends after each other and length of string checks) to be part of [staticcheck](https://github.com/dominikh/go-tools) instead, which we use in our CI tests for commits and pull requests.

Hi @Jacalz 

Thanks for the review! 

DeepSource takes care of more than 200+ Code Issues that might materialize as a Bug or an Anti-Pattern later on. It would thus save a lot of time on making the static checks stricter for new issues that would come up every now and then. 

It also runs on every commit and PR, which you can view from your integrated dashboard [here](https://deepsource.io/gh/HarshCasper/fyne/). With the Autofix, you can also get PR fixes automatically for common bugs and anti-patterns. 

Let me know what you think about the same!

File: widget/list_test.go [hunk 2]
```diff
@@ -279,9 +279,7 @@ func TestList_RemoveItem(t *testing.T) {
 	defer test.NewApp()
 
 	var data []string
-	data = append(data, "Test Item 0")
-	data = append(data, "Test Item 1")
-	data = append(data, "Test Item 2")
+	data = append(data, "Test Item 0", "Test Item 1", "Test Item 2")
```
Comment (Jacalz) [inline widget/list_test.go:282]: This can be added to the data variable att top directly without even needing an `append`.

**Review by Jacalz — CHANGES_REQUESTED** [review]
> Let me know what you think about the same!

Sorry, but my position has not changed. I think [staticcheck](https://github.com/dominikh/go-tools) is doing a terrific job already with providing very strict static analysis of our code and doing so commit and PR as well. It is also entirely open source which I think is a good plus. Adding more checks is also going to make checks run longer for PRs.

Added a small code recommendation as a further improvement to the changed code. I think it also would be better to base it on and target the `develop` branch where PRs are intended to be pointed to (especially as the code being touched has been changed in that branch.

**Comment by andydotxyz** [comment]
> It also runs on every commit and PR, which you can view from your integrated dashboard [here](https://deepsource.io/gh/HarshCasper/fyne/). With the Autofix, you can also get PR fixes automatically for common bugs and anti-patterns.

Hi @HarshCasper I agree with @Jacalz and have not seen sufficient value to address my earlier comment "am not sure I see enough in this change log to have another service running against our code for each commit."

If you are happy to remove the dependency on the new external system we can merge this PR. If you wish to leave the DeepSource link in this PR then we'll just need to close it until the product provides more value to a project such as ours.

**Comment by HarshCasper** [comment]
Thanks @Jacalz and @andydotxyz for the review! 

I will close this PR, since we have a new one here: https://github.com/fyne-io/fyne/pull/1979

**Pull request closed without merging**